### ISSUE TITLE
[watchOS] Need to register the UDID of paired Watches to the user's account

### ISSUE THREAD

**Issue by Matchstic** [Matchstic/ReProvision#42]
Paired Watches to the host iOS device are not registered to the user's account before downloading the appropriate provisioning profile for Watch applications.

This causes `installd` on the Watch to refuse the embedded provisioning profile for a given application, since the current device's UDID is not listed as a provisioned device within it. As a result, the application will fail to install, without an error on the iOS side.

**Comment by Matchstic** [comment]
To read the Apple Watch's UDID, we can use the following approach:

1. Import NanoRegistry.framework 
2. Add this entitlement to our binary: `com.apple.nano.nanoregistry` (value of `true`)
3. Call the following:
```
NRDevice *currentWatchDevice = [[NRPairedDeviceRegistry sharedInstance] getActivePairedDevice];
NSString *watchUDID = [currentWatchDevice valueForProperty:@"UDID"];
```

**Comment by Matchstic** [comment]
Implemented by 01da49e997db0168f4d5b8bd5004805f020912a3
